manuskript/ui/editors/editorWidget.py

```diff
@@ -344,12 +352,33 @@ def modelDataChanged(self, topLeft, bottomRight):
         if topLeft.row() <= self.currentIndex.row() <= bottomRight.row():
             self.updateStatusBar()
 
-    #def rowsAboutToBeRemoved(self, parent, first, last):
-        #if self.currentIndex:
-            #if self.currentIndex.parent() == parent and \
-                                    #first <= self.currentIndex.row() <= last:
-                ## Item deleted, close tab
-                #self.mw.mainEditor.tab.removeTab(self.mw.mainEditor.tab.indexOf(self))
+    def rowsInserted(self, parent, first, last):
+        if self.currentIndex:
+            self.updateIndexFromID(fallback=self.currentIndex.parent())
+
+    def rowsAboutToBeRemoved(self, parent, first, last):
+        if not self.currentIndex:
+            return
+
+        # Check parents until we find a parent that matches the parent we are removing.
+        crumb = self.currentIndex
+        commonality = crumb.parent() # start at folder above current text
```

Comment:
I'm not sure about `crumb` and `commonality` as variable names, I'd suggest `childItem` and `commonParent` 
respectively.. I don't like the names I found either but they are less confusing, if you have a better proposal, feel free to suggest it!

Reply:
I took a while to come up with these.

Commonality is because I wanted to stay away from the word parent and child because of the loaded meaning with both `QModelIndex` and our own model heavily using those terms. It made the code easier to read and understand for me to avoid those words. Besides: it is a perfectly fine english word. If you think the word isn't specific enough given that it may not actually be a commonality yet, how about `commonalityCandidate`? xD 

Crumb comes from the idea of a breadcrumb trail; we want to be able to look back at the item we originally came from.

Reply:
Yeah, I understood the commonality and crumb meaning, and yes they are valid english words, they are just confusing to me and not very clear on what they are for in this context, that's why I suggested changing them. If you prefer them to remain as is, that's your choice, feel free to ignore my comment :)

Reply:
Thank you both @worstje and @kakaroto for tackling code challenges such as the ones addressed by this PR.   When you are comfortable with the code in this PR then please let me know so that I can take another look at it with an eye towards merging it with the **develop** branch.

Reply:
@kakaroto Fact is my terms do indeed diverge from the terminology in the rest of the project, so you have a very valid point indeed. 😄 Others will have to understand the code later, so why should I stay stubborn?

But because I want to keep the code understandable and avoid the way 'parent' would be overly represented in that snippet, I have opted for a third option that uses the best of both worlds: `childItem` and `ancestorCandidate`. These should best describe the intent of the code in question. (I have also polished up the comments in this snippet of code to match the same language.)